### PR TITLE
Fallback to readline when launching with prompt toolkit 2.x installed

### PR DESCRIPTION
**Pull request by DNGros**
Rather than throwing an error, added functionality for falling back to readline shell when version 2.x of prompt toolkit is installed. Please see see discussion on #2570

### CODE REVIEW

File: xonsh/platform.py [hunk 1]
```diff
@@ -141,10 +141,14 @@ def ptk_version_info():
 
 
 @functools.lru_cache(1)
-def ptk_version_is_supported():
+def ptk_above_min_supported():
     minimum_required_ptk_version = (1, 0)
     return ptk_version_info()[:2] >= minimum_required_ptk_version
 
+@functools.lru_cache(1)
```
Comment (gforsyth) [inline xonsh/platform.py]: need another new line above this line to clear the flake failures

File: xonsh/platform.py [hunk 2]
```diff
@@ -141,10 +141,14 @@ def ptk_version_info():
...
+def ptk_below_max_supported():
+    ptk_max_version_cutoff = (2, 0)
+    return ptk_version_info()[:2] < ptk_max_version_cutoff
```
Comment (gforsyth) [inline xonsh/platform.py]: Similarly, need another new line after this one for flake failures

File: xonsh/shell.py [hunk 3]
```diff
@@ -132,11 +132,19 @@ def __init__(self, execer, ctx=None, shell_type=None, **kwargs):
...
                 warnings.warn('prompt-toolkit version < v1.0.0 is not '
                               'supported. Please update prompt-toolkit. Using '
                               'readline instead.')
                 shell_type = 'readline'
+            elif not ptk_below_max_supported():
+                warnings.warn('prompt-toolkit version 2.0 is not yet '
+                              'supported. Please see Github PR #2570 for '
+                              'latest status. To use prompt-toolkit now you '
+                              'can downgrade to version 1.x with\n'
+                              'pip install "prompt_toolkit<2"\n'
```
Comment (gforsyth) [inline xonsh/shell.py]: Can you change this to `xpip install "prompt_toolkit<2"`?
`xpip` is an alias to make sure that you call the correct version of `pip` to install it into the same env that `xonsh` resides in (mostly an issue for homebrew installs)

File: xonsh/platform.py (hunk 1 above)
Comment (DNGros) [reply xonsh/platform.py]: Sorry about that. I'll change it and keep this in mind in the future. Thanks for point it out.

**Comment by gforsyth** [comment]
Thanks @DNGros for helping to make our failures a little more graceful!

**Pull request merged**